src/components/organism/AddPoolLiquidity/index.tsx

```diff
@@ -371,6 +383,48 @@ const AddPoolLiquidity = ({ tokenBId }: AddPoolLiquidityProps) => {
     tokenBalances,
   ]);
 
+  const calculatePriceImpact = async () => {
+    if (api) {
+      if (selectedTokenNativeValue?.tokenValue !== "" && selectedTokenAssetValue?.tokenValue !== "") {
+        const poolSelected: any = pools?.find(
+          (pool: any) =>
+            pool?.[0]?.[1]?.interior?.X2?.[1]?.GeneralIndex?.replace(/[, ]/g, "") === selectedTokenB.assetTokenId
+        );
+        if (poolSelected && selectedTokenNativeValue?.tokenValue && selectedTokenAssetValue?.tokenValue) {
+          const poolReserve: any = await getPoolReserves(
+            api,
+            poolSelected?.[0]?.[1]?.interior?.X2?.[1]?.GeneralIndex?.replace(/[, ]/g, "")
+          );
+
+          const assetTokenReserve = formatDecimalsFromToken(
+            poolReserve?.[1]?.replace(/[, ]/g, ""),
+            selectedTokenB.decimals
+          );
+
+          const nativeTokenReserve = formatDecimalsFromToken(poolReserve?.[0]?.replace(/[, ]/g, ""), "12");
+
+          const priceBeforeSwap = new Decimal(nativeTokenReserve).div(assetTokenReserve);
+
+          const priceOfAssetBForOneAssetA = new Decimal(assetTokenReserve).div(nativeTokenReserve);
+          setAssetBPriceOfOneAssetA(priceOfAssetBForOneAssetA.toFixed(5));
+
+          const valueA = new Decimal(selectedTokenNativeValue?.tokenValue).add(nativeTokenReserve);
```

Comment:
```suggestion
          const valueA = new Decimal(selectedTokenNativeValue?.tokenValue || 0).add(nativeTokenReserve);
```
just to be safe and not get unhanded exception

Reply:
i don't think that his is necesary, we are already checking that it is not an empty string, it can be an empty string or a value bigger than 0